### ISSUE TITLE
[BUG] [3001] No matching sls found for ... in env base

### ISSUE THREAD

**Issue by remyd1** [saltstack/salt#57716]
**Description**
`salt-master` need to be restarted after calling a script with salt `3001`.
This script is committing /pushing my git remote fs code and is clearing cache, synchronizing module, functions... My code `clear_all.sh`, is available [here](https://gist.github.com/remyd1/2853f489d828ecda52183ca37a813f4e#file-clear_all-sh).

If I don't restart the master I get the message `No matching sls found for 'XXX' in env 'base'` for every states (which are present).

On the target side, I tested minion in 3001 and other versions, I still have the same error.

**Setup**

```bash
salt-call --versions-report
Salt Version:
           Salt: 3001
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.6.1
      docker-py: Not Installed
          gitdb: 2.0.3
      gitpython: 2.1.8
         Jinja2: 2.10
        libgit2: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.5.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: 3.4.7
         pygit2: Not Installed
         Python: 3.6.9 (default, Apr 18 2020, 01:56:04)
   python-gnupg: 0.4.1
         PyYAML: 3.12
          PyZMQ: 17.1.2
          smmap: 2.0.3
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.2.5
 
System Versions:
           dist: ubuntu 18.04 Bionic Beaver
         locale: UTF-8
        machine: x86_64
        release: 4.15.0-50-generic
         system: Linux
        version: Ubuntu 18.04 Bionic Beaver
```

**Steps to Reproduce the behavior**
Upgrade master to `3001`, then launch `clear_all.sh -g=y`.

**Expected behavior**
Having no issue with `No matching sls...`, and restarting `salt-master` should not be mandatory.

**Versions Report**
See above.

**Comment by xeacott** [comment]
@remyd1 Thanks for posting this, and agreed although we've seen a _few_ issues lately with needing to restart the `salt-master`. I am unsure exactly what best practice is, so maybe someone on @saltstack/team-core can weigh in on that?

**Comment by whytewolf** [comment]
Yes, a restart is going to be required if you rm the gitfs directories out of the cache. you not only deleted the repos you deleted the map to those repos. so while the master might think one thing salt-run is not going to be able to map to those items. 

That map is created on startup if it doesn't already exist.

**Comment by remyd1** [comment]
Ok @whytewolf , I will upgrade my scripts then. Is it related to a specific commit on salt `3001` ?

Thanks.

**Comment by whytewolf** [comment]
No, This should never have worked at all. If it had you were only getting lucky.

**Comment by remyd1** [comment]
The "rm" operation in my script is only done if `-e=y`. I don't do that usually. And it was not done in this use case.
